File: Soundmarine.lib/Model/TrackList.cs
using TagLib.Id3v2;

namespace Soundmarine.lib.Model;

public class TrackList
{
    public string PlaylistId { get; set; }
    public int TrackNumber { get; set; }
    public string TrackId { get; set; }
    public string DateAdded { get; set; }

    public TrackList(string playlistId, int trackNumber, string trackId, string dateAdded)
    {
        PlaylistId = playlistId;
        TrackNumber = trackNumber;
        TrackId = trackId;
        DateAdded = dateAdded;
    }
}

File: soundmarine_client/lib/models/track_list.dart
class TrackList {
  final String playlistId;
  final int trackNumber;
  final String trackId;
  final String dateAdded;

  TrackList({
    required this.playlistId,
    required this.trackNumber,
    required this.trackId,
    required this.dateAdded,
  });

  factory TrackList.fromJson(Map<String, dynamic> json) {
    return TrackList(
      playlistId: json['playlistId'],
      trackNumber: json['trackNumber'],
      trackId: json['trackId'],
      dateAdded: json['dateAdded'],
    );
  }
}